src/bin/leanc.in

```diff
@@ -19,14 +19,16 @@ bindir=$(dirname $0)
 cflags=("-I$bindir/../include")
 ldflags=("-L$bindir/../lib/lean" "${LEANC_GMP:--lgmp}" @LEANC_EXTRA_FLAGS@)
 ldflags_ext=(@LEANC_STATIC_LINKER_FLAGS@)
+args=("$@")
 for arg in "$@"; do
     # passed -shared ~> switch to shared linker flags
-    [[ $arg == "-shared" ]] && ldflags_ext=(@LEANC_SHARED_LINKER_FLAGS@)
-    [[ $arg == "-c" ]] && ldflags=() && ldflags_ext=()
+    [[ $arg == "-shared" ]] && ldflags_ext=(@LEANC_SHARED_LINKER_FLAGS@) && args=("-x" "c" "$@" "-x" "none")
```

Comment:
Readded the flags but only in compile mode. We don't need them for linking, and this way compilers which get confused when passed `-x` at link-time (as `emcc`) still work.